libs/barista-components/filter-field/src/filter-field.ts

```diff
@@ -827,6 +841,223 @@ export class DtFilterField<T = any>
+    const toPreviousFocusableTag =
+      (this._currentlyFocusedTag !== 'first' &&
+        this._currentlyFocusedTag === 'editButton') ||
+      this._currentlyFocusedTag === 'last';
+
+    if (toEditButton) {
+      return currentTag.editButton;
+    } else if (toPreviousFocusableTag) {
+      return this._getNextFocusableTagButton(
+        this._tags.toArray()[focusedTagIndex - 1],
+        true,
+      );
+    }
+  }
+
+  /** @internal Finds the next (subsequent) tag or input in default mode. */
+  private _findNextTagDefaultMode(
+    currentTag: DtFilterFieldTag | undefined,
+    focusedTagIndex: number,
+  ): ElementRef<HTMLElement> | undefined {
+    const toInputElement =
+      this._currentlyFocusedTag === 'last' ||
+      (this._currentlyFocusedTag === 'first' &&
+        this._tags.length === 1 &&
+        !this._hasDeleteButton(currentTag!));
+    const toDeleteButton =
+      (this._currentlyFocusedTag === 'editButton' ||
+        this._currentlyFocusedTag === 'first') &&
+      this._hasDeleteButton(currentTag!);
+    const toNextFocusableTag =
+      this._currentlyFocusedTag === 'editButton' ||
+      this._currentlyFocusedTag === 'deleteButton' ||
+      this._currentlyFocusedTag === 'first';
+
+    if (toInputElement) {
+      this._currentFocusState = 'input';
+      return this._inputEl;
+    } else if (toDeleteButton) {
+      return currentTag?.deleteButton;
+    } else if (toNextFocusableTag) {
+      return this._getNextFocusableTagButton(
+        this._tags.toArray()[focusedTagIndex + 1],
+        false,
+      );
+    }
+  }
+
+  /** @internal Sets the type based on the currently focused tag */
+  private _getCurrentlyFocusedTagType(currentTag: DtFilterFieldTag): void {
+    const firstTagHasFocus =
+      this._tags.first.editButton.nativeElement === document.activeElement;
+    const lastTagHasFocus =
+      this._getNextFocusableTagButton(this._tags.last, true).nativeElement ===
+      document.activeElement;
+    const editButtonHasFocus =
+      currentTag.editButton.nativeElement === document.activeElement;
+    const deleteButtonHasFocus =
+      this._getNextFocusableTagButton(currentTag, true).nativeElement ===
+      document.activeElement;
+
+    if (firstTagHasFocus) this._currentlyFocusedTag = 'first';
+    else if (lastTagHasFocus) this._currentlyFocusedTag = 'last';
+    else if (editButtonHasFocus) this._currentlyFocusedTag = 'editButton';
+    else if (deleteButtonHasFocus) this._currentlyFocusedTag = 'deleteButton';
+  }
+
+  /**
+   * @internal Returns a focusable tag element.
+   * The member deleteButtonFirst determines whether to check for the delete button first or not
+   */
+  _getNextFocusableTagButton(
+    tag: DtFilterFieldTag,
+    deleteButtonFirst: boolean,
+  ): ElementRef<HTMLButtonElement> {
+    if (deleteButtonFirst) {
+      if (tag.deleteButton) {
+        return tag.deleteButton;
+      } else {
+        return tag.editButton;
+      }
```

Comment:
Unnecessary else. As you already return in the if condition, you can do sth like this: 
```suggestion
        return tag.editButton;
```